### ISSUE TITLE
Configuration file support

### ISSUE THREAD

**Issue by tahir-hassan** [extrawurst/gitui#2140]
Whenever I run GitUI in a new working directory, when comparing differences, I have to do `o` to get into Options, then under Diff, I set "Ignore whitespaces" to `true`.

I would like it so that within the config directory, I can place a `gitui.ron` file with the content:

```ron
(
    diff: (
        ignore_whitespace: true,
        context: 3,
        interhunk_lines: 0,
    )
)
```

Then when GitUI starts up, it uses these values as the new defaults, and any changes you make would be written to `.git/gitui` in the working directory.

**Comment by remique** [comment]
Wait this is already implemented, right? I'm browsing through the code and I see that it saves the configuration under `.git/gitui` file.

**Comment by extrawurst** [comment]
No, what this issue is about is not yet implemented: falling back on some system wide set of options for gitui

**Comment by remique** [comment]
Ah, I misread what the issue was about 🥲

I am working on this one.
